bookmarks/services/parser.py

```diff
@@ -12,18 +14,81 @@ class NetscapeBookmark:
     tag_string: str
 
 
+class BookmarkParser(HTMLParser):
+
+    tag_stack: List[Optional[str]] = []
+    tag: Optional[str] = None
+    bookmarks: List[NetscapeBookmark] = []
+    bookmark: Optional[NetscapeBookmark] = None
+    folder: Optional[str] = ''
+    description: Optional[str] = None
+
+    def handle_starttag(self, tag: str, attrs: list):
+        name = 'handle_start_' + tag.lower()
+        if name in dir(self):
+            getattr(self, name)({k.lower(): v for k, v in attrs})
+        self.tag = tag
+
+    def handle_endtag(self, tag: str):
+        name = 'handle_end_' + tag.lower()
+        if name in dir(self):
+            getattr(self, name)()
+        self.tag = None
+
+    def handle_data(self, data):
+        name = f'handle_{self.tag}_data'
+        if name in dir(self):
+            getattr(self, name)(data)
+
+    def handle_start_dl(self, attrs: Dict[str, str]):
+        print('<DL>')
+        self.tag_stack.append(None)
+        self.description = None
+
+    def handle_end_dl(self):
+        print('</DL>')
+        self.add_bookmark()
+        self.tag_stack.pop()
+
+    def handle_start_dt(self, attrs: Dict[str, str]):
+        self.add_bookmark()
+
+    def handle_start_a(self, attrs: Dict[str, str]):
+        print(f'<A {attrs}>')
+        vars(self).update(attrs)
+
+    def handle_a_data(self, data):
+        print('<A>data:', data)
+        print(' self:', self.href, self.add_date)
+        print(' stack:', self.tag_stack)
+        self.bookmark = NetscapeBookmark(
+            href=self.href,
+            title=data,
+            description=self.description,
+            date_added=self.add_date,
+            tag_string=','.join(self.tag_stack[:-1]),
```

Comment:
I think what's missing here is also applying the tags from the anchor tag itself, currently the explicit tags on the anchor tag are ignored and only the tags created from folders are added.